Engine/media/audio/audio.cpp

```diff
@@ -634,23 +699,26 @@ void shutdown_sound()
 
 // the sound will only be played if there is a free channel or
 // it has a priority >= an existing sound to override
-int play_sound_priority (int val1, int priority) {
+static int play_sound_priority (int val1, int priority) {
     int lowest_pri = 9999, lowest_pri_id = -1;
 
+    AudioChannelsLock lock;
+
     // find a free channel to play it on
     for (int i = SCHAN_NORMAL; i < MAX_SOUND_CHANNELS; i++) {
+        auto* ch = lock.GetChannel(i);
         if (val1 < 0) {
             // Playing sound -1 means iterate through and stop all sound
             if (channel_is_playing(i))
                 stop_and_destroy_channel (i);
         }
         else if (!channel_is_playing(i)) {
             if (PlaySoundEx(val1, i) >= 0)
-                channels[i]->priority = priority;
+                ch->priority = priority;
```

Comment:
defect: This code will segfault because `ch` is null but it is expecting channels[i] to have a value.  There may be other examples to watch out for.

Reply:
And this is why in my old refactored branch I was returning locked *channels* instead of clips!